File: src (cpp)/animal/pisces/shark.cpp
#include "shark.h"

Shark::Shark(int _x, int _y, int _weight) : def_ID(13),
                                            def_ratio_meat(70),
                                            def_ratio_plant(0) {
  ID = def_ID;
  position.SetX(_x);
  position.SetY(_y);
  ratio_meat = def_ratio_meat;
  ratio_plant = def_ratio_plant;
  is_land_animal = false;
  is_water_animal = true;
  is_air_animal = false;
  weight = _weight;
}
void Shark::Interact() {
  std::cout << "The shark is staring at you menacingly" << std::endl;
}

File: src/virtualzoo/animal/pisces/Shark.java
package virtualzoo.animal.pisces;

import virtualzoo.animal.behavior.BehaviorTame;
import virtualzoo.animal.diet.Carnivora;

/**
 * Kelas Shark mendefinisikan atribut untuk shark.
 *
 * @author Holy Lovenia - 13515113
 * @version 3.0
 * @since 3.0
 */
public class Shark extends Pisces {

  /**
   * Constructor
   *
   * Menciptakan barracuda dengan posisi (locX,locY) dan berat animalWeight.
   *
   * @param locX Nilai absis posisi.
   * @param locY Nilai ordinat posisi.
   * @param animalWeight Nilai berat barracuda.
   */
  public Shark(int locX, int locY, int animalWeight) {
    super();
    animalBehavior = new BehaviorTame();
    animalDiet = new Carnivora(animalWeight, 70);
    ID = 13;
    position.setLocation(locX, locY);
    isLandAnimal = false;
    isWaterAnimal = true;
    isAirAnimal = false;
  }

  /**
   * Mengatur interaksi dengan shark.
   */
  public void interact() {
    System.out.println("The shark is staring at you menacingly");
  }
}
